.github/workflows/build-test-distribute.yaml

```diff
@@ -102,23 +128,40 @@ jobs:
         with:
           name: test-reports
           path: build/reports
-          retention-days: 30
+          retention-days: 7
   distributions:
-    needs: test
+    needs: ["check", "build", "test"]
     if: ${{ always() && !failure() && !cancelled() }}
     runs-on: ubuntu-latest
     steps:
+      - name: "Check if should run on all arch/os combinations"
+        if: github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'ci/run-full-matrix')
+        run: |
+          echo 'ENABLED_GOARCHES=arm64 amd64' >> $GITHUB_ENV
+          echo 'ENABLED_GOOSES=linux darwin' >> $GITHUB_ENV
+          # some steps don't support cross running, so write the actual arch down here.
+          ARCH=$(uname -m)
+          case $ARCH in
+            aarch64)
+            RUNNER_ARCH=arm64
+            ;;
+            x86_64)
+            RUNNER_ARCH=amd64
+            ;;
+            *)
+            echo "Unsupported arch '$ARCH'"
+            exit 1
+            ;;
+          esac
+          echo "RUNNER_ARCH=$RUNNER_ARCH" >> $GITHUB_ENV
```

Comment:
this is duplicated, can we dry it up?

Reply:
Unfortunately, they don't support setting environment variables across extracted actions.
This script will be removed because it's not needed anymore.